docs/en/explanation/implicit-indexes.rst

```diff
@@ -0,0 +1,46 @@
+Implicit indexes
+================
+
+So-called implicit indexes are named as such because they are created
+without the user asking for it. Some RDBMS will do so when you create a
+foreign key: they will create an index on the referencing table, using
+the referencing columns.
+
+The rationale behind this is that these indexes improve performance, for
+instance for checking that  a delete operation can be performed on a
+referenced table without violating the constraint in the referencing
+table.
+
+
+Here are some database platforms that are known to create indexes when
+creating a foreign key:
+
+- `MySQL <https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html>`_
+- `MariaDB <https://mariadb.com/kb/en/foreign-keys>`_
+
+
+Some other will not do so, on grounds that such indexes are not always
+needed, and can be created in many different ways. They instead leave
+that responsibility to the user:
+
+- `PostgreSQL <https://stackoverflow.com/questions/970562/postgres-and-indexes-on-foreign-keys-and-primary-keys>`_
+- `SQLite <https://sqlite.org/foreignkeys.html#fk_indexes>`_
+- `SQL Server <https://stackoverflow.com/questions/836167/does-a-foreign-key-automatically-create-an-index>`_
+
+In the case of these systems, the DBAL does the index creation for you
```

Comment:
Didn't we discover earlier that the DBAL will always create the necessary index w/o expecting the target platform to create an implicit one?

Reply:
We did, I have to rework this.